rockcraft/cli.py

```diff
@@ -66,6 +69,30 @@
 ]
 
 
+def run() -> None:
+    """Command-line interface entrypoint."""
+    # Register our own plugins
+    plugins.register()
+
+    # set lib loggers to debug level so that all messages are sent to Emitter
+    for lib_name in ("craft_providers", "craft_parts"):
+        logger = logging.getLogger(lib_name)
+        logger.setLevel(logging.DEBUG)
+
+    services = craft_application.ServiceFactory(
+        # type: ignore # type: ignore[call-arg]
+        app=APP_METADATA,
+        PackageClass=RockcraftPackageService,
+    )
+
+    app = Rockcraft(app=APP_METADATA, services=services)
+
+    try:
+        app.run()
+    except FallbackRunError:
+        legacy_run()
```

Comment:
One side-effect of doing it like this is that, by creating two Dispatchers, verbose modes get two "Starting rockcraft" greetings.

Reply:
No problem if this is temporary and goes away at the end of migration.